### PR TITLE
Load matrix configuration from workspace file

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Reads the list of locations to build from the `cloud-workspace.yaml` file instead of from the workflow file. This means most users won't need to touch the workflow file to configure their builds.

This is done by separating the CI workflow into two jobs: one which reads the locations to build, and then another which builds each location in parallel. 

The first step is completed by running some python code to read the YAML and output some JSON with build details. It does this in a lightweight composite action, meaning no need to download a (potentially bulky) Docker image. This lets the initial job complete quite quickly (~3 seconds).

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5** <a href="https://app.graphite.dev/github/pr/dagster-io/cloud-branch-deployments-action/5" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/cloud-branch-deployments-action/5?utm_source=stack-comment).

File: deploy/action.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ outputs:
     description: 'The Cloud deployment associated with this branch.'
 runs:
   using: 'docker'
-  image: 'docker://public.ecr.aws/a6x1l7i5/gh-action-test:0.30'
+  image: 'docker://public.ecr.aws/a6x1l7i5/gh-action-test:load-build-info'
```
Comment (benpankow) [inline deploy/action.yml:30]: needed for a CLI tweak, will be removed before merge

File: .github/workflows/branch_deployments.yml [hunk 2]
```diff
@@ -6,18 +6,30 @@ env:
   DAGSTER_CLOUD_URL: https://pied-piper.dogfood.dagster.cloud
 
 jobs:
+  # Reads the list of locations to build from the cloud workspace file
```
Comment (prha) [inline .github/workflows/branch_deployments.yml:9]: These are just example files, to copy as in a template, yes?

**Review by prha — APPROVED** [review]

File: .github/workflows/branch_deployments.yml (hunk 2 above)
Comment (benpankow) [reply .github/workflows/branch_deployments.yml:9]: yeah, examples/"tests" - will put up a corresponding change to the quickstart repo as well (that's the one people fork at the moment )

**Pull request merged**